source/common/ssl/context_impl.cc

```diff
@@ -351,8 +358,14 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
       ASN1_STRING* str = san->d.uniformResourceIdentifier;
       const char* uri = reinterpret_cast<const char*>(ASN1_STRING_data(str));
       for (auto& config_san : subject_alt_names) {
-        if (config_san.compare(uri) == 0) {
-          return true;
+        if (is_san_regex_validation) {
+          if (std::regex_match(config_san, regex_pattern)) {
```

Comment:
This is matching the wrong field: the regex should be matched against `uri`, not against the configuration-supplied `config_san`.